src/rpc/rawtransaction.cpp

```diff
@@ -169,6 +170,19 @@ void TxToJSON(const CTransaction& tx, const uint256 hashBlock, UniValue& entry)
             o.push_back(Pair("asm", ScriptToAsmStr(txin.scriptSig, true)));
             o.push_back(Pair("hex", HexStr(txin.scriptSig.begin(), txin.scriptSig.end())));
             in.push_back(Pair("scriptSig", o));
+
+            // Add address and value info if spentindex enabled
+            CSpentIndexValue spentInfo;
+            CSpentIndexKey spentKey(txin.prevout.hash, txin.prevout.n);
+            if (GetSpentIndex(spentKey, spentInfo)) {
+                in.push_back(Pair("value", ValueFromAmount(spentInfo.satoshis)));
+                in.push_back(Pair("valueSat", spentInfo.satoshis));
+                if (spentInfo.addressType == 1) {
```

Comment:
These really should be enums or constants. I see in #3837 the `Type()` method was added to abstract over the numeric constants from the upstream patchset; it should be modified similarly (and since that is stored data, if using an enum it would need to be typed appropriately in order to maintain backwards-compatibility).

Alternatively, add a function that lives right next to where these numbers are defined (the `Type()` function I guess?) that does this switching, and returns a `boost::option` containing the result of the respective `EncodeDestination` call, or `boost::none` if it is an unsupported type. Then this would become something like:
```suggestion
                auto address = GetTransparentAddressForSpentInput(spentInfo);
                if (address) {
                    in.push_back(Pair("address", address.get()));
                }
```